### PR TITLE
Added new tests for shipment-types BAPI endpoint. Remove tags for tests where bug has been fixed. 

### PR DESCRIPTION
**Pull request by Nataliia-Popkova**
## PR Description

TBD

## Checklist
- [x] I agree with the Code Contribution License Agreement in CONTRIBUTING.md

### CODE REVIEW

File: resources/steps/shipment_type_steps.robot [hunk 1]
```diff
@@ -0,0 +1,41 @@
+@@ -0,0 +1,209 @@
+*** Settings ***
+Library    DatabaseLibrary
+Library    BuiltIn
+Library    ../../resources/libraries/common.py
+Resource    ../common/common_api.robot
+
+*** Keywords ***
+
+Get shipment type id in DB by key:
```
Comment (AntonShopin) [inline resources/steps/shipment_type_steps.robot:10]: in DB vs from DB?

File: tests/api/suite/bapi/shipment_types/negative.robot [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+*** Test Cases ***
+ENABLER
+    TestSetup
+
+*** Test Cases ***
+Create_shipment_type_with_incorrect_type_in_body
+    [Documentation]    https://spryker.atlassian.net/browse/FRW-5845
+    [Tags]    skip-due-to-issue
+    [Setup]    Run Keywords    I get access token by user credentials:   ${zed_admin.email}
+    ...    AND    I set Headers:    Content-Type=application/vnd.api+json   Authorization=Bearer ${token}
```
Comment (AntonShopin) [inline tests/api/suite/bapi/shipment_types/negative.robot:17]: `Content-Type=application/vnd.api+json`
It's better to move `application/vnd.api+json` into a separate variable, smth like ${default_bapi_header_content_type}

File: tests/api/suite/bapi/shipment_types/negative.robot [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+    And Response should return error code:    5501
+    And Response should return error message:    A delivery type entity was not found.
+
+Create_shipment_type_with_empty_key_in_request
+    [Setup]    Run Keywords    I get access token by user credentials:   ${zed_admin.email}
+    ...    AND    I set Headers:    Content-Type=application/vnd.api+json   Authorization=Bearer ${token} 
+    When I send a POST request:    /shipment-types        {"data": {"type": "shipment-types","attributes": {"name": "Some Shipment Type ${random}","key": "","isActive": "true","stores": ["DE", "AT"]}}}
+    Then Response status code should be:    400
+    And Response should return error code:    5503
+    And Response should return error message:    	A delivery type key must have a length from 1 to 255 characters.
```
Comment (AntonShopin) [inline tests/api/suite/bapi/shipment_types/negative.robot:64]: Too many whitespaces between the keyword and argument/value

File: resources/common/common_api.robot [hunk 4]
```diff
@@ -935,11 +935,44 @@ Response should contain the array larger than a certain size:
     @{data}=    Get Value From Json    ${response_body}    ${json_path}
     Log    @{data}
     ${list_length}=    Get Length    @{data}
+    Log    ${list_length}
```
[the remark this answers is not in the document]
Comment (Nataliia-Popkova) [reply resources/common/common_api.robot]: removed

File: resources/common/common_api.robot [hunk 5]
```diff
@@ -935,11 +935,44 @@ Response should contain the array larger than a certain size:
...
     ${list_length}=    Convert To Integer    ${list_length}
     ${result}=    Evaluate   ${list_length} > ${expected_size}
     ${result}=    Convert To String    ${result}
     Should Be Equal    ${result}    True    Actual array length is '${list_length}' and it is not greater than expected '${expected_size}' in '${json_path}'.
 
+Each array element of array in response should contain the array larger than a certain size:
```
[the remark this answers is not in the document]
Comment (Nataliia-Popkova) [reply resources/common/common_api.robot]: fix keyword, remove the not needed loop. Change variable names and test name

File: resources/common/common_api.robot [hunk 6]
```diff
@@ -935,11 +935,44 @@ Response should contain the array larger than a certain size:
...
+    [Documentation]    This keyword checks that each element in the array specified as ``${json_path}`` contains the `` ${expected_array}`` with certain size greater than ``${expected_size}``.
+    ...
+    ...    If at least one array element has ``${expected_array}`` grater than ``${expected_size}``, the keyword will fail.
+
+    ...    *Example:*
+    ...
+    ...    `` Each array element of array in response should contain the array larger than a certain size:    [data]    [attributes][stores]    0``
+    [Arguments]    ${json_path}    ${expected_array}    ${expected_size}
```
[the remark this answers is not in the document]
Comment (Nataliia-Popkova) [reply resources/common/common_api.robot]: fix keyword, remove the not needed loop. Change variable names and test name

File: resources/common/common_api.robot [hunk 7]
```diff
@@ -935,11 +935,44 @@ Response should contain the array larger than a certain size:
...
+    ${list_length}=    Get Length    @{data}
+    ${list_length}=    Get From List    @{data}    ${index}
+    @{data}=    Get Value From Json    ${list_length}    ${expected_array}
+    ${list_length_new}=    Get Length    @{data}
+        FOR    ${index}    IN RANGE    0    ${list_length_new}
+    ${list_length_new}=    Convert To Integer    ${list_length_new}
+    Log    ${list_length_new}
+    ${result}=    Evaluate   ${list_length_new} > ${expected_size}
+    ${result}=    Convert To String    ${result}
+    Should Be Equal    ${result}    True    Each array '${list_length_new}' is not greater than expected '${expected_size}' in '${json_path}'.
```
[the remark this answers is not in the document]
Comment (Nataliia-Popkova) [reply resources/common/common_api.robot]: change error message

File: resources/common/common_api.robot [hunk 8]
```diff
@@ -935,11 +935,44 @@ Response should contain the array larger than a certain size:
...
+    # Should Not Be Equal    ${result}    False    At leat one ${expected_array} length greater than expected '${expected_size}' in '${json_path}'.
+        END
+    END 
+    
+
+
+
```
[the remark this answers is not in the document]
Comment (Nataliia-Popkova) [reply resources/common/common_api.robot]: removed

File: resources/common/common_api.robot [hunk 9]
```diff
@@ -935,11 +935,44 @@ Response should contain the array larger than a certain size:
...
+    @{data}=    Get Value From Json    ${response_body}    ${json_path}
+    ${list_length}=    Get Length    @{data}
+    ${log_list}=    Log List    @{data}
+    FOR    ${index}    IN RANGE    0    ${list_length}
+    @{data}=    Get Value From Json    ${response_body}    ${json_path}
+    ${list_length}=    Get Length    @{data}
+    ${list_length}=    Get From List    @{data}    ${index}
+    @{data}=    Get Value From Json    ${list_length}    ${expected_array}
+    ${list_length_new}=    Get Length    @{data}
+        FOR    ${index}    IN RANGE    0    ${list_length_new}
```
[the remark this answers is not in the document]
Comment (Nataliia-Popkova) [reply resources/common/common_api.robot]: aligned

File: resources/common/common_api.robot [hunk 10]
```diff
@@ -1069,14 +1102,64 @@ Each array element of array in response should contain property with value in:
...
+    ${list_length}=    Get Length    @{data}
+    ${log_list}=    Log List    @{data}
+    FOR    ${index}    IN RANGE    0    ${list_length}
+        ${list_element}=    Get From List    @{data}    ${index}    
+        ${list_element}=    Get Value From Json    ${list_element}    ${expected_property}
+        ${list_element}=    Convert To String    ${list_element}
+        ${list_element}=    Replace String    ${list_element}    '   ${EMPTY}
+        ${list_element}=    Replace String    ${list_element}    [   ${EMPTY}
+        ${list_element}=    Replace String    ${list_element}    ]   ${EMPTY}  
+    Should Not Be Empty     ${list_element}    '${json_path}' property value '${expected_property}' is empty but shoud not be
```
[the remark this answers is not in the document]
Comment (Nataliia-Popkova) [reply resources/common/common_api.robot]: fixed

File: resources/steps/shipment_type_steps.robot [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+    [Documentation]    This keyword get the entry from the DB table `spy_shipment_type`.
```
[the remark this answers is not in the document]
Comment (Nataliia-Popkova) [reply resources/steps/shipment_type_steps.robot]: fixed

File: resources/steps/shipment_type_steps.robot (hunk 1 above)
Comment (Nataliia-Popkova) [reply resources/steps/shipment_type_steps.robot]: changed

File: tests/api/suite/bapi/shipment_types/negative.robot (hunk 2 above)
Comment (Nataliia-Popkova) [reply tests/api/suite/bapi/shipment_types/negative.robot]: added variable

File: tests/api/suite/bapi/shipment_types/positive.robot [hunk 12]
```diff
@@ -0,0 +1,162 @@
...
+    And Response body parameter should not be EMPTY:    [data][id]
+    And Response body parameter should be:    [data][attributes][name]    Some Shipment Type ${random}
+    And Response body parameter should be:    [data][attributes][key]    some-shipment-type-${random}
+    And Response body parameter should be:    [data][attributes][isActive]    True
+    And Response body parameter should be in:    [data][attributes][stores]    DE    AT
+    And Response body parameter should be in:    [data][attributes][stores]    AT    DE
+    And Response body has correct self link for created entity:    ${shipment_type_id}
+    [Teardown]     Delete shipment type in DB:    some-shipment-type-${random}
+
+Create_new_shipment_type_with_existing_Not-Uniqu_Name_unique_key_and_IsActive
```
[the remark this answers is not in the document]
Comment (Nataliia-Popkova) [reply tests/api/suite/bapi/shipment_types/positive.robot]: changed to new one

File: tests/api/suite/bapi/shipment_types/positive.robot [hunk 13]
```diff
@@ -0,0 +1,162 @@
...
+    And Save value to a variable:    [data][id]    shipment_type_id
+    And Response body parameter should not be EMPTY:    [data][id]
+    And Response body parameter should be:    [data][attributes][name]    not_unnique_name
+    And Response body parameter should be:    [data][attributes][key]    second-shipment-type-${random}
+    And Response body parameter should be:    [data][attributes][isActive]    True
+    And Response body parameter should be in:    [data][attributes][stores]    DE    AT
+    And Response body parameter should be in:    [data][attributes][stores]    AT    DE
+    And Response body has correct self link for created entity:    ${shipment_type_id}
+    [Teardown]     Run Keywords    Delete shipment type in DB:    new-shipment-type-${random}
+     ...    AND    Delete shipment type in DB:    second-shipment-type-${random}
```
[the remark this answers is not in the document]
Comment (Nataliia-Popkova) [reply tests/api/suite/bapi/shipment_types/positive.robot]: add additional assertion

File: tests/api/suite/bapi/shipment_types/positive.robot [hunk 14]
```diff
@@ -0,0 +1,162 @@
...
+    When I send a POST request:    /shipment-types
+    ...    {"data": {"type": "shipment-types","attributes": {"name": "name${random}","key": "update-shipment-type-key${random}","isActive": "true","stores": ["DE"]}}}
+    Then Response status code should be:    201
+    And Save value to a variable:    [data][id]    shipment_type_uuid
+    # Update the Delivery Type with new attributes via PATCH request
+    When I send a PATCH request:    /shipment-types/${shipment_type_uuid}
+    ...    {"data": {"type": "shipment-types","attributes": {"name": "updated_name${random}","isActive": "false","stores": ["AT"]}}} 
+    Then Response status code should be:    200
+    And Response reason should be:    OK
+    And Response body parameter should be:    [data][attributes][name]    updated_name${random}
```
[the remark this answers is not in the document]
Comment (Nataliia-Popkova) [reply tests/api/suite/bapi/shipment_types/positive.robot:70]: add Get request after the PATCH

**Comment by Nataliia-Popkova** [comment]
Checked in CI https://app.travis-ci.com/github/spryker/suite-nonsplit/builds/265433696

**Comment by Nataliia-Popkova** [comment]
https://app.travis-ci.com/github/spryker/suite-nonsplit/builds/265433696 issue with push notification will be fixed separatly.

**Comment by Nataliia-Popkova** [comment]
last run are green https://github.com/spryker/suite-nonsplit/runs/16212508028

**Review by AntonShopin — APPROVED** [review]

**Pull request merged**